dvc/ignore.py

```diff
@@ -172,14 +175,38 @@ def __init__(self, fs, root_dir):
         self.root_dir = root_dir
         self.ignores_trie_fs = Trie()
         self._ignores_trie_subrepos = Trie()
+        self.config = Config()
 
         key = self._get_key(root_dir)
+
+        core_config = self.config.get("core", {})
+        config_ignore_file = core_config.get("excludesfile", None)
+
+        def extend_default_ignore_patterns(ignore_file_path):
+            ignore_patterns = DvcIgnorePatterns.from_file(
+                ignore_file_path, self.fs, ignore_file_path
+            )
+            default_ignore_patterns.extend(ignore_patterns.pattern_list)
+
+        if config_ignore_file:
+            if self.fs.exists(config_ignore_file):
+                extend_default_ignore_patterns(config_ignore_file)
+        else:
+            for level in ["global", "system"]:
+                ignore_file_path_at_level = self.fs.path.join(
+                    Config.get_dir(level), DvcIgnore.DVCIGNORE_FILE
+                )
+                if self.fs.exists(ignore_file_path_at_level):
+                    extend_default_ignore_patterns(ignore_file_path_at_level)
+                    break
```

Comment:
How about extracting this to a separate (class)method that returns the list ignore patterns read from the config file?

Reply:
Refactored this code to be as suggested